lightningd/peer_htlcs.c

```diff
@@ -994,12 +994,27 @@ void peer_sending_commitsig(struct peer *peer, const u8 *msg)
 		      take(towire_channel_sending_commitsig_reply(msg)));
 }
 
-static void added_their_htlc(struct peer *peer,
-			     const struct added_htlc *added,
-			     const struct secret *shared_secret)
+static bool peer_added_their_htlc(struct peer *peer,
+				  const struct added_htlc *added,
+				  const struct secret *shared_secret)
 {
 	struct htlc_in *hin;
 
+	/* BOLT #2:
+	 *
+	 *  - receiving an `amount_msat` equal to 0, OR less than its own `htlc_minimum_msat`:
+	 *    - SHOULD fail the channel.
+	 */
+	if (added->amount_msat < 0
```

Comment:
how would this ever be true if `amount_msat` is an unsigned integer?

Reply:
Seems clang caught this one too:

```
lightningd/peer_htlcs.c:1008:25: error: comparison of unsigned expression < 0 is
      always false [-Werror,-Wtautological-compare]
        if (added->amount_msat < 0
            ~~~~~~~~~~~~~~~~~~ ^ ~
1 error generated.
```